pkg/applicationserver/io/packages/loragls/v3/api/objects.go

```diff
@@ -119,7 +119,7 @@ func BuildSingleFrameRequest(ctx context.Context, metadata []*ttnpb.RxMetadata)
 		Frame:    Frame{},
 	}
 	for _, m := range metadata {
-		if m.Location == nil {
+		if m.Location == nil || m.GetGatewayIds() == nil {
```

Comment:
No need for the getter here as if `m == nil`, `m.Location` will panic anyway. Same below.

Reply:
Ok I'll remove the getter but `m.GatewayIds` can still be nil even if `m != nil`.

Reply:
You're correct. It's just that there is no point to use the getter if we access the other fields without it.